### PR TITLE
NAS-103620 / 11.3 / fix(middlewared/download): do not close pipes for running download

### PR DESCRIPTION
**Pull request by william-gr**
Also increase read timeout for downloads.

Ticket:	NAS-103620

### CODE REVIEW

File: src/middlewared/middlewared/main.py [hunk 1]
```diff
@@ -377,7 +377,14 @@ def register_job(self, job_id):
         self.jobs[job_id] = self.middleware.loop.call_later(
             60, lambda: asyncio.ensure_future(self._cleanup_job(job_id)))
 
+    async def _cleanup_cancel(self, job_id):
+        job = self.jobs.pop(job_id, None)
```
Comment (themylogin) [inline src/middlewared/middlewared/main.py]: Took me a while to understand what's going on. Can we name it `job_cleanup`? For me it looked like we are canceling the job itself.

File: src/middlewared/middlewared/main.py [hunk 2]
```diff
@@ -439,9 +446,10 @@ def do_copy():
                 asyncio.run_coroutine_threadsafe(resp.write(read), loop=self.loop).result()
 
         try:
+            await self._cleanup_cancel(job_id)
             await self.middleware.run_in_thread(do_copy)
         finally:
-            await self._cleanup_job(job_id)
+            await job.pipes.close()
```
Comment (themylogin) [inline src/middlewared/middlewared/main.py]: Code duplication with `_cleanup_job`

File: src/middlewared/middlewared/main.py (hunk 1 above)
Comment (william-gr) [reply src/middlewared/middlewared/main.py]: We already have a _cleanup_job method. Why would we call it job_cleanup if its really canceling a future?

Comment (themylogin) [reply src/middlewared/middlewared/main.py]: @william-gr I mean name variable `job_cleanup`

**Pull request merged**